scripts/automation/includes/github-api-optimized.js

```
@@ -0,0 +1,373 @@
+/**
+ * Fetches all items from a paginated endpoint
+ * @param {string} path - API path with query parameters
+ * @param {object} options - Request options including per_page
+ * @returns {Promise<object[]>} All items from all pages
+ */
+export async function fetchPaginated(path, options = {}) {
+  const allItems = [];
+  const perPage = options.perPage || 30;
+  let page = 1;
+
+  while (true) {
```

ESLint

[error] Unexpected constant condition.

(no-constant-condition)